### ISSUE TITLE
One-time charge complaining about webhook URL

### ISSUE THREAD

**Issue by wiwa1978** [mollie/laravel-cashier-mollie#36]
Building a Laravel app with one time charge. I use the following code, based upon [this](https://www.cashiermollie.com/04-charges.html#one-time-charges) doc.
```
$user = Auth::user();

$item = new ChargeItemBuilder($user);
        $item->unitPrice(money(100,'EUR')); 
        $item->description('Test Item 1');
        $chargeItem = $item->make();

        $result = $user->newCharge()
            ->addItem($chargeItem)
            ->create();

 if(is_a($result, RedirectToCheckoutResponse::class)) {
            return $result;
        }
```
In the `config/cashier.php` I have set the `webhook_url`. In my case it points to an expose tunnel (similar to ngrok). 

```
    'webhook_url' => 'https://13uls2e2tc.sharedwithexpose.com/webhooks/mollie',
```
The tunnel is working. The webhooks/mollie route is also existing and is being handled by the controller.

I always get the following error:

`Error executing API call (422: Unprocessable Entity): The webhook URL is invalid because it is unreachable from Mollie's point of view.. Field: webhookUrl.`

Note: i have been doing the same with the Mollie PHP API ([here](https://github.com/mollie/mollie-api-php)) and that works well.

**Comment by Mul-tiMedia** [comment]
As far as I can see it uses the `cashier.first_payment.webhook_url` instead of `cashier.webhook_url`. Could you verify this?

**Comment by sandervanhooft** [comment]
@wiwa1978 you can try this to help you debug and get a quick overview:

```php
dd([
    'cashier.first_payment.webhook_url' => config('cashier.first_payment.webhook_url'),
    'cashier.webhook_url' => config('cashier.webhook_url'),
]);
```

One thing that I've noticed in the past is that if you switch the app url in `.env` to the tunnel's address, and then point the browser to for example `http://localhost:8080/mollietest`, Laravel uses the localhost address as the app url. This results in a local - unreachable - webhook address. You can prevent this by pointing the browser also to the tunnel's address.

Hope this helps

**Comment by sandervanhooft** [comment]
Closing this for now, let me know if it should be reopened.

**Comment by wiwa1978** [comment]
Found the issue: for future reference, here's what I did:
- change the APP_URL in the .env file to the tunnel endpoint (e.g. https://13uls2e2tc.sharedwithexpose.com in the example above.
- despite this change in the .env file, Laravel does not use this tunnel endpoint in the URL. Instead it always keeps using http://localhost:8000. So added the following in the `AppServiceProvider` (to force Laravel to use the tunnel endpoint)

```    
public function boot() {
        URL::forceRootUrl(Config::get('app.url'));
 }
```
- using the above changes, it works perfectly
